domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -1194,6 +1195,40 @@ class ProfileManagementControllerTest {
     assertThat(lastSelectedClassroomId).isEqualTo(TEST_CLASSROOM_ID_1)
   }
 
+  @Test
+  fun testFetchLastSelectedClassroomId_updateClassroomIdTwice_checkUpdatesAreSuccessful() {
+    setUpTestApplicationComponent()
+    addTestProfiles()
+
+    monitorFactory.ensureDataProviderExecutes(
+      profileManagementController.loginToProfile(PROFILE_ID_0)
+    )
+    monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.updateLastSelectedClassroomId(
+        PROFILE_ID_0,
+        TEST_CLASSROOM_ID_0
+      )
+    )
+    val lastSelectedClassroomId0 = monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.retrieveLastSelectedClassroomId(PROFILE_ID_0)
+    )
+    assertThat(lastSelectedClassroomId0).isEqualTo(TEST_CLASSROOM_ID_0)
+
+    monitorFactory.ensureDataProviderExecutes(
+      profileManagementController.loginToProfile(PROFILE_ID_0)
+    )
+    monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.updateLastSelectedClassroomId(
+        PROFILE_ID_0,
+        TEST_CLASSROOM_ID_1
+      )
+    )
```

Comment:
This set up is kind of similar to the one for switching profiles. What has not been tested and is needed here is the scenario where the user in tha same profile switches classrooms e.g select math, then science. So the set up is log in once, update twice with different classrooms, and asert once that the second classroom is the latest value.

Reply:
Ah, I understand now, updated the test accordingly.